test/common/storagelib.py

```diff
@@ -610,8 +610,7 @@ def card_row(self, title: str, index: int | None = None, name: str | None = None
             return pfx + f" tbody [data-test-row-location='{location}']"
 
     def click_card_row(self, title: str, index: int | None = None, name: str | None = None, location: str | None = None, table_index: int = 1) -> None:
-        # We need to click on a <td> element since that's where the handlers are...
-        self.browser.click(self.card_row(title, index, name, location, table_index) + " td:nth-child(1)")
+        self.browser.click(self.card_row(title, index, name, location, table_index) + " .indent button")
```

Comment:
I don't like using `.indent` here as it mixes selectors that are made for UI purposes with tests which are supposed to be separate from UI changes. We can improve it _a bit_ by using the table cell selector to get to the button itself like this:
```suggestion
        self.browser.click(self.card_row(title, index, name, location, table_index) + "td:nth-child(1) button")
```
And even better, we can add a `data-ouia-component-id` to the Button and select it through that. [OUIA being a test selector framework](https://www.patternfly.org/developer-resources/open-ui-automation)
```suggestion
        self.browser.click(self.card_row(title, index, name, location, table_index) + f"[data-ouia-component-id='storage-link-cell'] [data-ouia-component-id='{location}']")
```

(Haven't tested this suggestion)

Reply:
Interesting, I had no idea what OUIA was! Thanks!

This is a significant topic for our tests in general, how do we identify DOM elements and how do we interact with them. This is done in a variety of ad-hoc ways, and our unwritten rule seems to be "whatever works, and don't worry about maintenance too much".

My change here is a good example: we want to click on the thing that's in the "main" column, and that is the column is also used for indenting, we use the class used for that.  If that ever breaks (once we get around to indenting "properly"), we will find out and have to change this one place in storagelib.py. Not a bit deal and not worth worrying about too much, it happened in this PR and it was not a problem, IMO.

But the fact that PF supports OUIA probably means that we should move towards it.

The Storage page in particular uses similar ideas to OUIA already. Table rows are identified via `data-test-row-name` and `data-test-row-location`, and dialogs use `data-field` and `data-field-type` to get pretty good abstractions for writing tests.

We could rename the `data-field-type` and `data-field` attributes used with the Storage dialogs to `data-ouia-component` and `data-ouia-component-id`, but is that worth it? It's just a change in name, with no real gain. (I especially wouldn't want to commit to OUIA compliance.)  `data-test-row-name` and `data-test-row-location` don't map cleanly to OUIA, and it's not clear to me how adapting it would improve things.

Our biggest issue is that we use things like `button:contains(Mount)`  a lot. I don't see much wrong with that in principle, but we need Sizzle to implement `:contains` and can't use `document.querySelector`.

So we could start with that, what do you think?

We could have a `click_button` function in testlib/Browser that could be used like this:

```
    b.click_button("apply")
```

This would expand into `b.click("[data-oui-component=PF6/Button][data-ouia-component-id=apply]")`. There is no way I would want to write or read "data-ouia-component-id" a million times in our tests. :-) I don't even want to write/read "click_ouia", that's not a word.

To allow more precision, `click_button` would take a optional prefix selector:

```
    b..click_button("apply", "#dialog")
```

This would expand into `b.click("#dialog [data-oui-component=PF6/Button][data-ouia-component-id=apply]")`, obviously.

This could be the start of a PF6 automation library, and we could also have functions for selecting from dopdown menus (we have that already, I think) and toggling checkboxes etc.

So we could name this `pf_click` instead of `click_button`, and have `pf_select_menu`, etc.

Or we could just have functions for building selectors:

```
    b.click(pf_button("apply"))
```

We could put these into their own file even. testlib.py is getting too crowded. But functions that do multiple steps and are not just building a selector are probably best part of the Browser class.

So this PR could do maybe this:

```
    b.pf_click("id-link", self.card_row(...))
```

and the code for creating the button would be

```
    <Button ... ouiaId="id-link">
```

The "context" for this button is the table row, in my view, and we need to distinguish between the link in the "Id" column and the link in the "Location" column (which would get a ouiaID="location") and the dropdown toggle ("actions").

Alright! So much for the Wednesday morning brain dump. :-) What do you think?